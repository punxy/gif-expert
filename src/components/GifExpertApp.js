import React, { useState } from 'react'
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

export const GifExpertApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState(defaultCategories)
   
    // const handleAdd = () => {
    //     //setCategories([ ...categories, 'HunterXHunter' ]);
    //     setCategories( cats =>  [...cats, 'HunterXHunter']);
    // }

    return (
        <>
            <h2>GifExpertApp</h2>
            <hr />

            <AddCategory setCategories={setCategories} />

            <ol>
                { categories.map( category => 
                        <GifGrid key={category} category={category} />
                )}
            </ol>
        </>
    )
}
