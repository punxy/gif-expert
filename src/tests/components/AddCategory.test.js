import React from 'react';
import { shallow } from "enzyme"
import '@testing-library/jest-dom';
import { AddCategory } from '../../components/AddCategory'

describe('Test in AddCategory component', () => {

    const setCategories = jest.fn();
    let wrapper = shallow( <AddCategory setCategories={setCategories} /> );

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={setCategories} /> );
    });

    test('should be render a component correctly',() => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should be change inputBox', () => {

        const value = 'Hola Mundo!';
        const input = wrapper.find('input');

        input.simulate('change', {
            target: { value }
        });

        const input2 = wrapper.find('input');
        expect(input2.prop('value')).toBe(value);
    });

    test('Should NOT update new values on the submit', () => {
        wrapper.find('form').simulate('submit', {
            preventDefault: () => {},
        });

        expect( setCategories ).not.toHaveBeenCalled();
    });

    test('should be call to setCategories and clear the input text', () => {
        
        const value = 'texto de prueba';
        wrapper.find('input').simulate('change', { target: { value } });
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect( setCategories ).toHaveBeenCalled();
        expect( setCategories ).toHaveBeenCalledTimes(1);
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function ) )

        expect( wrapper.find('input').prop('value') ).toBe('');

    })
});