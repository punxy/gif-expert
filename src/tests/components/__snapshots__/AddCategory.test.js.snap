// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in AddCategory component should be render a component correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
